app/src/components/main/index.tsx

```diff
@@ -29,23 +29,28 @@ import { MarketWizardCreatorContainer } from '../market/sections/market_create/m
 import { MarketHomeContainer } from '../market/sections/market_list/market_home_container'
 
 const RedirectToHome = () => <Redirect to="/" />
-
 export const Main: React.FC = () => {
   const context = useWeb3Context()
   const windowObj: any = window
-
-  const [networkId, setNetworkId] = useState(windowObj.ethereum.chainId)
+  const host = window.location.hostname
+  let defaultChainID = 1
+  if (host === XDAI_LOCATION) {
+    defaultChainID = 100
+  }
+  const [networkId, setNetworkId] = useState(windowObj.ethereum ? windowObj.ethereum.chainId : defaultChainID)
```

Comment:
One question why are we using defaultChainId variable and then networkId useState hook...why not just consolidate it to use hook and use setNetworkId to set default ones?

Reply:
NetworkID is not the same as chainid, they have the same values but they are not the same entity.
https://ethereum.stackexchange.com/questions/37533/what-is-a-chainid-in-ethereum-how-is-it-different-than-networkid-and-how-is-it